### PR TITLE
feat: [CDS-37893]:ssh-winrm-services

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary

[[](https://harness.atlassian.net/browse/CDS-37893)]
added new deployment types like 'ssh' and 'winrm', and functionality to select secret and local files, for config file
(https://harness.atlassian.net/browse/CDS-37893)

#### Screenshots
Screenshot 2022-07-19 at 11 48 34

Screenshot 2022-07-19 at 11 48 52
Screenshot 2022-07-19 at 11 48 59
Screenshot 2022-07-19 at 11 49 23

Screenshot 2022-07-17 at 20 00 10

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>
- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>

<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx [hunk 1]
```diff
@@ -220,8 +220,8 @@ const NewFileForm: React.FC<NewFileModalData> = props => {
       initialValues={initialValues}
       formName="newFile"
       validationSchema={Yup.object().shape({
-        identifier: IdentifierSchema(),
-        fileUsage: Yup.string().nullable(true).trim().required(getString('filestore.errors.fileUsage'))
+        identifier: IdentifierSchema()
+        // fileUsage: Yup.string().nullable(true).trim().required(getString('filestore.errors.fileUsage'))
```
Comment (bhavya-sinha) [inline src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx:224]: Commented?

File: src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx [hunk 2]
```diff
@@ -242,6 +242,14 @@ const NewFileForm: React.FC<NewFileModalData> = props => {
                 />
                 <FormInput.Select
                   style={{ width: 180 }}
```
Comment (bhavya-sinha) [inline src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx:244]: Inline style?

File: src/modules/31-filestore/components/MultiTypeFileSelect/EncryptedSelect/EncryptedFileSelectField.tsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ import css from './EncryptedFileSelectField.module.scss'
 interface SelectEncryptedProps {
   name: string
   formik: FormikContextType<any>
-  onChange: (newValue: any) => void
+  onChange: (newValue: any, id?: any) => void
```
Comment (bhavya-sinha) [inline src/modules/31-filestore/components/MultiTypeFileSelect/EncryptedSelect/EncryptedFileSelectField.tsx:19]: id can be string | number

File: src/modules/31-filestore/components/StoreView/StoreView.tsx [hunk 4]
```diff
@@ -31,7 +31,14 @@ export default function StoreView(): React.ReactElement {
   }
 
   return (
-    <Container style={{ width: '100%', height: 'calc(100vh - 75px)', overflow: 'hidden', overflowY: 'scroll' }}>
+    <Container
+      style={{
```
Comment (bhavya-sinha) [inline src/modules/31-filestore/components/StoreView/StoreView.tsx:35]: Inline style?

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactListView/ArtifactListView.tsx [hunk 5]
```diff
@@ -270,7 +271,7 @@ function ArtifactListView({
             text={getString('pipelineSteps.serviceTab.artifactList.addPrimary')}
           />
         )}
-        {isAdditionAllowed && isSidecarAllowed && (
+        {isAdditionAllowed && isSidecarAllowed && withSideCar && (
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ArtifactsSelection/ArtifactListView/ArtifactListView.tsx:274]: There was already a function `isSidecarAllowed` doing same thing. Check if you can use that

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesHelper.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+  if (fileType === FILE_TYPE_VALUES.FILE_STORE && isRunTime) {
+    filesData.files = formData?.files
+  }
+
+  if (fileType === FILE_TYPE_VALUES.ENCRYPTED && !isRunTime) {
+    filesData.secretFiles = formData?.files
+  }
+  if (fileType === FILE_TYPE_VALUES.ENCRYPTED && isRunTime) {
+    filesData.secretFiles = formData?.files
+  }
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesHelper.ts:115]: All 4 above if conditions can be combined

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+  const [isEditMode, setIsEditMode] = useState(false)
+  const [configStore, setConfigStore] = useState<ConfigFileType>('' as ConfigFileType)
+  const [configFileIndex, setEditIndex] = useState(0)
+  const [isNewFile, setIsNewFile] = useState(true)
+
+  const { expressions } = useVariablesExpression()
+
+  const commonProps = {
+    name: getString('credentials'),
+    onConnectorCreated: () => {
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx:103]: We can use noop here

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx [hunk 8]
```diff
@@ -0,0 +1,476 @@
...
+          set(draft, path, listOfConfigFiles)
+        }).stage as StageElementConfig
+      )
+    }
+  }
+
+  const handleSubmit = (configFileObj: ConfigFileWrapper): void => {
+    if (isPropagating) {
+      if (listOfConfigFiles?.length > 0) {
+        listOfConfigFiles.splice(configFileIndex, 1, configFileObj)
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx:163]: Should not try to mutate props. Props are readonly

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx [hunk 9]
```diff
@@ -0,0 +1,476 @@
...
+      return buildGithubPayload
+    }
+    if (type === Connectors.BITBUCKET) {
+      return buildBitbucketPayload
+    }
+    if (type === Connectors.GITLAB) {
+      return buildGitlabPayload
+    }
+    return () => ({})
+  }
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx:232]: There is exported func in `getBuildPayload` in ManifestHelper. You can try to use it here

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx [hunk 10]
```diff
@@ -0,0 +1,476 @@
...
+    setIsNewFile(false)
+    setConfigStore(configFileType)
+    setSelectedConfig(configFileType)
+    setNewConnectorView(false)
+    setEditIndex(index)
+    showConnectorModal()
+  }
+
+  const removeConfigFileConfig = (index: number): void => {
+    listOfConfigFiles.splice(index, 1)
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx:323]: props should be readonly

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx [hunk 11]
```diff
@@ -0,0 +1,476 @@
...
+              {getString('pipelineSteps.serviceTab.manifestList.manifestStore')}
+            </Text>
+            <Text font={{ variation: FontVariation.TABLE_HEADERS }}>{getString('location')}</Text>
+            <span></span>
+          </div>
+        )}
+        <Layout.Vertical style={{ flexShrink: 'initial' }} width="100%">
+          <section>
+            {listOfConfigFiles &&
+              listOfConfigFiles.map((data: ConfigFileWrapper, index: number) => {
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx:406]: Chaining can be used

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/ConfigFilesStore.tsx [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+  stepName,
+  isReadonly,
+  configFilesStoreTypes,
+  initialValues,
+  // previousStep,
+  expressions,
+  allowableTypes,
+  prevStepData,
+  nextStep
+}: // isNewFile
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/ConfigFilesStore.tsx:72]: Please remove commented codes

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesWizard.tsx [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+      title={`${ConfigWizard.label} ${getString('pipeline.configFiles.title', { type: 'Source' })}`}
+      initialStep={1}
+    >
+      <ConfigFilesStore
+        configFilesStoreTypes={types}
+        name={getString('pipeline.configFiles.source')}
+        stepName={getString('pipeline.configFiles.source')}
+        // name={'Config File Details'}
+        // stepName={'Config File Details'}
+        // selectedConfigFile={selectedConfigFile}
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesWizard.tsx:59]: Remove commented code

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/DeployServiceSpecifications.tsx [hunk 14]
```diff
@@ -91,7 +92,8 @@ export default function DeployServiceSpecifications({
     scope,
     getStageFromPipeline,
     updateStage
-  } = usePipelineContext()
+  } = context
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/DeployServiceSpecifications.tsx:95]: Why this change?

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

File: scripts/strings/yamlStringsCheck.mjs [hunk 15]
```diff
@@ -61,10 +61,7 @@ function validateReferences(str, path, restricedModules = [], isOld) {
 
 function validateStrings(data, parentPath = [], restricedModules = [], isOld) {
   if (!data) {
-    errors.push([
```
Comment (kapilkumawat86) [inline scripts/strings/yamlStringsCheck.mjs:64]: Is this change intended

**Comment by kapilkumawat86** [comment]
dockerize

File: scripts/strings/yamlStringsCheck.mjs (hunk 15 above)
Comment (litvinovstas) [reply scripts/strings/yamlStringsCheck.mjs:64]: hmm, i didn't modified this file.. don't understand why it's changed

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx [hunk 16]
```diff
@@ -0,0 +1,476 @@
...
+  }
+  const commonLastStepProps = {
+    handleSubmit,
+    expressions
+  }
+  const getLastSteps = useCallback((): Array<React.ReactElement<StepProps<any>>> => {
+    const arr: Array<React.ReactElement<StepProps<any>>> = []
+    let configDetailStep = null
+
+    switch (true) {
```
[the remark this answers is not in the document]
Comment (litvinovstas) [reply src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx]: resolved

File: src/modules/31-filestore/components/FileStoreComponent/FileStoreComponent.tsx [hunk 17]
```diff
@@ -176,9 +176,13 @@ const useFileStoreModal = ({ applySelected }: UseFileStoreModalProps): UseFileSt
...
   }, [activeTab, selectedFile])
 
+  const handleShowModal = React.useCallback(() => {
+    showModal()
+  }, [])
+
   return {
     openFileStoreModal: () => {
-      showModal()
+      handleShowModal()
```
[the remark this answers is not in the document]
Comment (litvinovstas) [reply src/modules/31-filestore/components/FileStoreComponent/FileStoreComponent.tsx]: resolved

File: src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (litvinovstas) [reply src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx]: resolved

File: src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/DeployServiceSpecifications.tsx (hunk 14 above)
Comment (litvinovstas) [reply src/modules/75-cd/components/PipelineStudio/DeployServiceSpecifications/DeployServiceSpecifications.tsx]: resolved

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/ConfigFilesStore.tsx (hunk 12 above)
Comment (litvinovstas) [reply src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesWizard/ConfigFilesSteps/ConfigFilesStore.tsx]: resolved

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx (hunk 8 above)
Comment (litvinovstas) [reply src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesListView/ConfigFilesListView.tsx:165]: resolve

File: src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesHelper.ts (hunk 6 above)
Comment (litvinovstas) [reply src/modules/70-pipeline/components/ConfigFilesSelection/ConfigFilesHelper.ts:102]: resolved

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactsSelection.tsx [hunk 18]
```diff
@@ -679,6 +680,7 @@ export default function ArtifactsSelection({
       refetchConnectors={refetchConnectorList}
       isReadonly={readonly}
       isAdditionAllowed={isAdditionAllowed(deploymentType, readonly)}
+      withSideCar={!withoutSideCar(deploymentType)}
```
[the remark this answers is not in the document]
Comment (litvinovstas) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactsSelection.tsx]: resolved

File: src/modules/70-pipeline/components/ArtifactsSelection/ArtifactListView/ArtifactListView.tsx (hunk 5 above)
Comment (litvinovstas) [reply src/modules/70-pipeline/components/ArtifactsSelection/ArtifactListView/ArtifactListView.tsx]: resolved

File: src/modules/31-filestore/components/MultiTypeFileSelect/EncryptedSelect/EncryptedFileSelectField.tsx (hunk 3 above)
Comment (litvinovstas) [reply src/modules/31-filestore/components/MultiTypeFileSelect/EncryptedSelect/EncryptedFileSelectField.tsx]: resolved

File: src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx (hunk 2 above)
Comment (litvinovstas) [reply src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx]: resolved

File: src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx (hunk 1 above)
Comment (litvinovstas) [reply src/modules/31-filestore/common/useNewNodeModal/views/NewFileModalForm.tsx]: resolved

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/KubernetesServiceSpecInputSetMode.tsx [hunk 19]
```diff
@@ -91,6 +93,23 @@ const KubernetesServiceSpecInputSetModeFormikForm = (props: KubernetesInputSetPr
         />
       )}
 
+      {!!template?.configFiles?.length && (
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/KubernetesServiceSpecInputSetMode.tsx:96]: Looks like this needs to be removed yet.

**Review by chetanvekariya-harness — CHANGES_REQUESTED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**